### PR TITLE
Make org settings routes and nav items injectable

### PR DESCRIPTION
**Pull request by eseliger**
This adds the same pattern we use for user settings, repo settings etc. to inject enterprise-only routes and nav items into the area.

I will use this in a follow-up PR for an enterprise-only page that I need to add to the orgs page.

## Test plan

Verified the org pages still work as before.

## App preview:

- [Web](https://sg-web-es-custom-org-settings-rotues.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-jghzyvurun.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.14% (+3.98 kb) | -0.05% (-7.55 kb) | -0.10% (-11.53 kb) 🔽 | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/es/custom-org-settings-rotues/compare-report.html;tab=live_object) for a full comparison between the commits 4e00faa and dc60978 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/org/settings/OrgSettingsSidebar.tsx [hunk 1]
```diff
@@ -3,31 +3,62 @@ import * as React from 'react'
...
     const siteAdminViewingOtherOrg = authenticatedUser && org.viewerCanAdminister && !org.viewerIsMember
+    const context: OrgSettingsSidebarItemConditionContext = {
+        batchChangesEnabled: props.batchChangesEnabled,
+        batchChangesExecutionEnabled: props.batchChangesExecutionEnabled,
+        batchChangesWebhookLogsEnabled: props.batchChangesWebhookLogsEnabled,
+        org,
+        authenticatedUser,
+        isSourcegraphDotCom: props.isSourcegraphDotCom,
+        newMembersInviteEnabled,
+    }
```
Comment (courier-new) [inline client/web/src/org/settings/OrgSettingsSidebar.tsx:61]: It looks like the only `context` prop actually used by any of the items in `orgSettingsSideBarItems` is `newMembersInviteEnabled`. Could we drop the rest?

**Review by courier-new — APPROVED** [review]
👍

File: client/web/src/org/settings/OrgSettingsSidebar.tsx (hunk 1 above)
Comment (eseliger) [reply client/web/src/org/settings/OrgSettingsSidebar.tsx:61]: I think they'll be used when I start adding things, so I'm a bit hesitant to remove them rn given they've been useful on the user settings and we've been using them there (which is sort of what I will mirror here). Happy to drop them later if they don't prove useful!

Comment (courier-new) [reply client/web/src/org/settings/OrgSettingsSidebar.tsx:61]: Fair enough, sounds good! 🙂

**Pull request merged**